s3torchconnectorclient/rust/src/logger_setup.rs

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * // SPDX-License-Identifier: BSD
+ */
+use std::{env, io::Write};
+use std::str::FromStr;
+use env_logger::{Builder, Target};
+use log::{LevelFilter};
+use mountpoint_s3_crt::common::rust_log_adapter::RustLogAdapter;
+use pyo3::{PyResult};
+use chrono::Local;
+use pyo3_log::Logger;
+use crate::exception::python_exception;
+
+pub const ENABLE_CRT_LOGS_ENV_VAR: &str = "ENABLE_CRT_LOGS";
+const DATE_FORMAT_STR: &str = "%Y-%m-%d %H:%M:%S%.3f";
+
+pub fn setup_logging() -> PyResult<()> {
+    let enable_crt_logs = env::var(ENABLE_CRT_LOGS_ENV_VAR)
+        .unwrap_or(String::from("OFF"))
+        .to_uppercase();
+
+    match enable_crt_logs.as_str() {
+        "OFF" => enable_default_logging(),
+        level_filter_str => enable_crt_logging(level_filter_str)?
+    }
```

Comment:
The problem is that we cannot combine `RustLogAdapter` (which brings the CRT logs on Rust side) with `pyo3-log`, because we run into the pyo3 deadlock issue and we do not control what is happening on the CRT side. 
The naming is confusing, I will update that. 

`enable_crt_logging` configures the logging such that both CRT and MP logs are handled by Rust logging, and are not sent to Python logger anymore. This might result in a format difference, as the format is not inherited from Python, but we can live with that for debugging purposes.

`enable_default_logging()` is logging as usual, sending MP logs from Rust to Python through pyo3, and is used if the environment variable is not set / set to "off". The CRT logs are not processed.